x/lockup/keeper/lock.go

```diff
@@ -48,18 +48,22 @@ func (k Keeper) BeginUnlockAllNotUnlockings(ctx sdk.Context, account sdk.AccAddr
 }
 
 // AddToExistingLock adds the given coin to the existing lock with the same owner and duration.
-// Returns an empty array of period lock when a lock with the given condition does not exist.
-func (k Keeper) AddToExistingLock(ctx sdk.Context, owner sdk.AccAddress, coin sdk.Coin, duration time.Duration) ([]types.PeriodLock, error) {
+// Returns the updated lock ID if successfully added coin, returns 0 when a lock with
+// given condition does not exist.
+func (k Keeper) AddToExistingLock(ctx sdk.Context, owner sdk.AccAddress, coin sdk.Coin, duration time.Duration) (uint64, error) {
 	locks := k.GetAccountLockedDurationNotUnlockingOnly(ctx, owner, coin.Denom, duration)
 	// if existing lock with same duration and denom exists, just add there
 	if len(locks) > 0 {
+		// there should only be a single lock with the same duration + token, thus we take the first lock
```

Comment:
Please consider adding a check for lock length not being 1, returning an error in such a case and a test.

I think it's useful to make sure that we catch errors as soon as possible. For example, if someone accidentally introduces a bug in `GetAccountLockedDurationNotUnlockingOnly`, returning no locks and there is no test for it, we would be able to catch it right here.

Reply:
I was also thinking about this as well, I think further investigation needs to be done before we change the logic that you've pointed out, since this was a logic that has been laying here for a long time.

I wanted to investigate more on this and confirm with certainty before we change it so that we assume that lock length is always 1. I can create an issue to track this! lmk what you think

Reply:
If that's the case, then wouldn't we have to come to a conclusion on whether lock length = 1 is a sound assumption before merging this?

Happy to help investigate if you get blocked/want a second pair of eyes

Reply:
This was the exact logic gate(we had the same check before this PR) we had before changing so that part should be fine in terms of safteyness

Reply:
https://github.com/osmosis-labs/osmosis/issues/2142